### ISSUE TITLE
Find Artifacts From Resource overrides docker image baked by helm3 

### ISSUE THREAD

**Issue by mkj28** [spinnaker/spinnaker#6506]
### Issue Summary:
The container image tag in the baked manifest is ignored if there's "find artifacts from resource" earlier in the pipeline - and the image tag from the "find artifacts" is deployed.

Seems to have broken between 1.22.1 (was not an issue) and 1.24.6 (happens here)

### Cloud Provider(s):
Kubernetes

### Environment:
Spinnaker 1.24.6

### Feature Area:
Pipelines

### Description:
image.tag overwritten by the one from the "Find Artifacts From Resource (Manifest) Configuration" step - ignoring the image.tag baked by helm3 engine

### Steps to Reproduce:
1. Have "Find Artifacts From Resource (Manifest) Configuration" step in the pipeline that fetches kubernetes deployment information
2. Bake helm chart that includes value for image to be deployed (.spec.containers.image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}") and image.tag in Overrides
3. Confirm baked deployment yaml has the expected image.tag (it does)
3. Run the pipeline with deployment to kubernetes

Expected: 
image.tag in deployment from baked deployment yaml used

Actual:
image.tag overwritten by the one from the "Find Artifacts From Resource (Manifest) Configuration" step

### Additional Details:
Example ("find artifacts from resources" has image: "myapp:v1):
Baked Manifest YAML has (truncated):
```
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    tag: "v2"
spec:
    spec:
      containers:
          image: "myapp:v2"
```
but on the deployment step the deployment yaml has (truncated):
```
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    tag: "v2"
spec:
    spec:
      containers:
          image: "myapp:v1"
```

![2021-08-11 _174920](https://user-images.githubusercontent.com/2279299/129122401-70326c38-49a8-46da-be76-d91b2a14228a.png)

---

**_Please delete the instructions below this line prior to submitting_**

**Comment by sikevux** [comment]
This sounds like the behavior added in spinnaker/clouddriver#4874 that was made into an option with spinnaker/clouddriver#5343

**Comment by mkj28** [comment]
> This sounds like the behavior added in [spinnaker/clouddriver#4874](https://github.com/spinnaker/clouddriver/pull/4874) that was made into an option with [spinnaker/clouddriver#5343](https://github.com/spinnaker/clouddriver/pull/5343)

indeed. I guess looks like will be available in 1.27. Thanks!
